testsuite/MDAnalysisTests/lib/test_cutil.py

```diff
@@ -46,6 +48,14 @@ def test_unique_int_1d(values):
     assert res.dtype == ref.dtype
 
 
+def test_array_unique_int_1d():
+    try:
+        res_indices = np.array([6, 5, 5, 7], dtype=np.int32)
+        assert_equal(unique_int_1d(res_indices),  [5, 6, 7])
```

Comment:
I know @orbeckst suggested this, but a dtype of `np.int_` would be more representative of the `resindices` array across Windows and Unix systems, as it is created for most topology formats with the dtype `np.int`, i.e. `np.int_`. [`np.int_` is either an int32 or int64 depending on the platform](https://numpy.org/devdocs/user/basics.types.html), which is likely why this has not been an issue earlier.

https://github.com/MDAnalysis/mdanalysis/blob/b341090aaa0bace9b946316ebf1e5bf965b026da/package/MDAnalysis/topology/base.py#L203

As it is this test is guaranteed to fail from a type mismatch.

Reply:
Yes i will surely make the change and update the PR

Reply:
I wasn't sure what the right choice for the `dtype` should have been, so by all means, follow @lilyminium 's suggestion.

On a more general note: should we avoid the platform-dependent types in MDA and be explicitly using the numpy-guaranteed types instead?

Reply:
> should we avoid the platform-dependent types in MDA

Probably, to the extent reasonable

Reply:
I'm a little unclear on the suggestion to use `np.int_` here. It seems like we only have a problem when `np.int32` is used because of the platform, and it seems like we'd want to test for that regardless of what Appveyor happens to use for it.

So, if Appveyor does someday not use `np.int32` for `np.int_` anymore, we'd effectively stop testing for the regression in our CI.

Perhaps just parametrize over the two and then both people are right :)

Reply:
> So, if Appveyor does someday not use np.int32 for np.int_ anymore, we'd effectively stop testing for the regression in our CI.

I don't really follow -- aren't the platform differences (and the fact that apparently no one has tried this in Windows until now) the source of the bug? I would rather keep this test and stick an `xfail` over it to track changes in the Appveyor types. Passing an int32 array to a function that wants an int64 array should always fail, right, even on Unix, and probably not provide useful information.

Anyway, oops, looking at `change_squash()` was too upstream; the arrays get converted to `np.intp` in TransTable.

https://github.com/MDAnalysis/mdanalysis/blob/b341090aaa0bace9b946316ebf1e5bf965b026da/package/MDAnalysis/core/topology.py#L219

Reply:
We want to simulate the presence of *any* problematic platform that might substitute `np.int32` in `np.int_`, so I think always using `np.int32` makes sense to catch the regression consistently.

The currently-proposed fix that I see is just going to coerce to `np.int64` without a platform-specific guard, which seems right, since that can be made a no-op with `astype` when the input is already `int64`.

If you really want to decorate the test on POSIX machines, you could skip it when not windows, but probably not worth it--it should pass on all platforms really.

Reply:
But using `np.int32` here will make the test fail on all platforms -- the test passes the array directly into `unique_int_1d`, without trying to convert it to int64. Unless I'm misunderstanding and that's the point.

Reply:
Right, I think the test should probably be removed since `unique_int_1d` is not public?

It is on a private import path: `MDAnalysis.lib._cutil.unique_int_1d`

Or do you want users to be able to use it directly? Then the test is "ok" and we'd need a lower-level patch there, but I suspect that's intended to be private and the test should be removed.

If you agree, that would mean we're back to the stage that CI is not able to reproduce the failure for the first test, which is closer to the reported issue. For that I suspect we need a 32-bit entry in CI, but I don't know if we want to go down that rabbit hole based on the other PR/discussions there. I think it may be a good idea, but it is a maintenance burden to consider.

I'll see if I can reproduce locally now with a 32-bit win interpreter.

Reply:
That was a very good point though, I didn't actually look close enough to notice that the proposed patch wouldn't have an effect for the second test.